### PR TITLE
fix: update css for modals

### PR DESCRIPTION
**Pull request by sahar-fehri**
## **Description**
Few css updates on nft and token detection modal

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25961?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

### **Before**

image

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by salimtb — DISMISSED** [review]

File: ui/components/app/assets/auto-detect-nft/auto-detect-nft-modal.tsx [hunk 1]
```diff
@@ -117,15 +114,17 @@ function AutoDetectNftModal({ isOpen, onClose }: AutoDetectNftModalProps) {
...
           cancelButtonProps={{
             children: t('notRightNow'),
             block: true,
+            variant: ButtonVariant.Link,
+          }}
+          submitButtonProps={{
+            children: t('allow'),
+            block: true,
           }}
+          containerProps={{ flexDirection: FlexDirection.ColumnReverse }}
```
Comment (georgewrmarshall) [inline ui/components/app/assets/auto-detect-nft/auto-detect-nft-modal.tsx:127]: Added these updates @sahar-fehri 

Screenshot 2024-07-22 at 10 02 39 AM

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! This is great use of design system components pushed a small update to prevent the need to update design system components. It also looks like a design pattern that we may want to promote so I'll create an issue to better document this in Figma and storybook. I really appreciate you reaching out and the before/after screenshots in the PR description. Thank you!

File: ui/components/app/assets/auto-detect-nft/auto-detect-nft-modal.tsx (hunk 1 above)
Comment (sahar-fehri) [reply ui/components/app/assets/auto-detect-nft/auto-detect-nft-modal.tsx:127]: Thanks a lot @georgewrmarshall ! The  `flexDirection: FlexDirection.ColumnReverse` is way better 🙏

File: ui/components/app/assets/auto-detect-nft/auto-detect-nft-modal.tsx [hunk 2]
```diff
@@ -96,13 +97,9 @@ function AutoDetectNftModal({ isOpen, onClose }: AutoDetectNftModalProps) {
           >
             <img src="/images/wallet-alpha.png" />
           </Box>
-          <Text
-            variant={TextVariant.bodyMd}
-            textAlign={TextAlign.Justify}
-            padding={0}
-          >
+          <Text variant={TextVariant.bodyMd} padding={0}>
```
Comment (georgewrmarshall) [inline ui/components/app/assets/auto-detect-nft/auto-detect-nft-modal.tsx:100]: non-blocking: the Text component doesn't have any padding by default so I don't believe this is doing anything
```suggestion
          <Text variant={TextVariant.bodyMd}>
```

File: ui/components/app/assets/auto-detect-token/auto-detect-token-modal.tsx [hunk 3]
```diff
@@ -101,13 +100,9 @@ function AutoDetectTokenModal({
           >
             <img src="/images/wallet-alpha.png" />
           </Box>
-          <Text
-            variant={TextVariant.bodyMd}
-            textAlign={TextAlign.Justify}
-            padding={0}
-          >
+          <Text variant={TextVariant.bodyMd} padding={0}>
```
Comment (georgewrmarshall) [inline ui/components/app/assets/auto-detect-token/auto-detect-token-modal.tsx:103]: non-blocking: `Text` component has no padding by default I don't believe padding={0} is needed here
```suggestion
          <Text variant={TextVariant.bodyMd}>
```

File: ui/components/app/assets/auto-detect-token/auto-detect-token-modal.tsx [hunk 4]
```diff
@@ -132,8 +127,10 @@ function AutoDetectTokenModal({
           onCancel={() => handleTokenAutoDetection(false)}
           cancelButtonProps={{
             children: t('notRightNow'),
+            style: { borderStyle: 'none' },
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/app/assets/auto-detect-token/auto-detect-token-modal.tsx:130]: Apologies, I missed this in my initial review. Could we use the same approach as the auto-detect-nft-modal?

Import `ButtonVariant` from the component-library folder:
```suggestion
            variant: ButtonVariant.Link,
```

**Review by georgewrmarshall — COMMENTED** [review]
Re-reviewed, I missed the auto-detect-token modal on initital review. Left one suggestion

**Review by georgewrmarshall — COMMENTED** [review]
Apologies again for further changes, but on third review, I noticed that we may want to duplicate the entire approach from the auto-detect-nft `ModalFooter` as it's likely intentional to have the same pattern on both modals.

Screenshot 2024-07-22 at 11 25 33 AM
Screenshot 2024-07-22 at 11 25 52 AM

**Review by georgewrmarshall — APPROVED** [review]
LGTM!
- checked `AutoDetectNftModal` button positioning in storybook ✅ 
- checked `AutoDetectTokenModal` button positioning in storybook ✅

**Review by bergeron — APPROVED** [review]

**Pull request merged**